lib/ansible/modules/cloud/amazon/ec2_vpc_dhcp_options_facts.py

```diff
@@ -83,91 +84,75 @@
     returned: always
 '''
 
-import json
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import ec2_argument_spec, boto3_conn, HAS_BOTO3
+from ansible.module_utils.ec2 import ansible_dict_to_boto3_filter_list, get_aws_connection_info
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict, boto3_tag_list_to_ansible_dict
 
 try:
     import botocore
-    import boto3
-    HAS_BOTO3 = True
 except ImportError:
-    HAS_BOTO3 = False
+    pass  # caught by imported HAS_BOTO3
 
 
 def get_dhcp_options_info(dhcp_option):
     dhcp_option_info = {'DhcpOptionsId': dhcp_option['DhcpOptionsId'],
-                       'DhcpConfigurations': dhcp_option['DhcpConfigurations'],
-                       'Tags': dhcp_option['Tags']
-               }
+                        'DhcpConfigurations': dhcp_option['DhcpConfigurations'],
+                        'Tags': boto3_tag_list_to_ansible_dict(dhcp_option['Tags'])}
     return dhcp_option_info
 
 
 def list_dhcp_options(client, module):
-    dryrun = module.params.get("DryRun")
-    all_dhcp_options_array = []
-    params = dict()
-
-    if module.params.get('filters'):
-        params['Filters'] = []
-        for key, value in module.params.get('filters').items():
-            temp_dict = dict()
-            temp_dict['Name'] = key
-            if isinstance(value, basestring):
-                temp_dict['Values'] = [value]
-            else:
-                temp_dict['Values'] = value
-            params['Filters'].append(temp_dict)
-
-    if module.params.get("DryRun"):
-        params['DryRun'] = module.params.get("DryRun")
-
-    if module.params.get("DhcpOptionsIds"):
-        params['DhcpOptionsIds'] = module.params.get("DhcpOptionsIds")
+    params = dict(Filters=ansible_dict_to_boto3_filter_list(module.params.get('filters')))
+
+    if module.params.get("dry_run"):
+        params['DryRun'] = module.params.get("dry_run")
```

Comment:
Shouldn't this just be `params['DryRun'] = True` since the dry_run property is a boolean, so it'll only hit this when it's true anyway?